snuba/web/delete_query.py

```diff
@@ -0,0 +1,36 @@
+from snuba.clickhouse.formatter.query import format_query
+from snuba.clickhouse.query import Query
+from snuba.datasets.storages.storage_key import StorageKey
+from snuba.query.conditions import combine_and_conditions
+from snuba.query.data_source.simple import Table
+from snuba.query.dsl import column, equals, in_cond, literal, literals_tuple
+
+
+def construct_condition(body):
+    and_conditions = []
+    for col, values in body.get("columns").items():
+        if len(values) == 1:
+            exp = equals(column(col), literal(values[0]))
+        else:
+            literal_values = [literal(v) for v in values]
+            exp = in_cond(column(col), literals_tuple(literal_values))
+
+        and_conditions.append(exp)
+
+    return combine_and_conditions(and_conditions)
+
+
+def delete_query(storage: StorageKey, table: str, body):
+    query = Query(
```

Comment:
@xurui-c I think we can call processors on this directly since it would be in the correct form. And also for the `MaxRowEnforcer` I think it should be possible to take the `from_clause` and `condition` and construct your own `Query` by adding your select, but I didn't test this out